src/Model/NotificationModel.php

```diff
@@ -13,39 +13,133 @@
 
 class NotificationModel implements NotificationInterface
 {
-    private ?string $url = null;
+    private string $id;
 
-    public function __construct(private string $id, private string $message, private string $type = Constants::TYPE_INFO)
+    // Simple link
+    private ?string $message       = null;
+    private ?string $type          = null;
+    private ?string $url           = null;
+    private bool    $active        = false;
+    private bool    $disabled      = false;
+    private bool    $withBadge     = true;
+    private bool    $badgeAnimated = true;
+
+    // Custom HTML
+    private ?string $html = null;
+
+    public function isValid(): bool
     {
+        return $this->message || $this->html;
+    }
+
+    public function __construct(
+        string $id,
```

Comment:
Constructor property promotion! This is PHP 8... 😁 but seriously, not that old-school redudancy please

Reply:
Yeah I know 😞
I really don't like to use those when we have properties IN class + constructor.
Look a bit messy for me.
But I use it a lot with dependency injection!

Anyway, I'll change that.